### ISSUE TITLE
can i train my model with your amazing library on cpu?

### ISSUE THREAD

**Issue by marcusau** [jidasheng/bi-lstm-crf#1]
Thanks for your fantastic piece of work.

May I ask if this library can be run on cpu computer?  

I would like to train my model on my office desktop which has no gpu. 

Thanks a lot.

**Comment by jidasheng** [comment]
Sure! I think it works though I don't test it in a non-GPU machine. @_@

**Comment by marcusau** [comment]
Thanks , i am trying it now :)

**Comment by jidasheng** [comment]
Looking forward to your results!

**Comment by marcusau** [comment]
two more questions... as i am about to start my training .

1. Usually, we use BMES format for chinese word segmentation,  why you use BI format instead??  any advantage of it???

2. there are char-based and word-based in your corpus format.... how to tell the train.py which form we input for training? 


Thanks a lot

**Comment by jidasheng** [comment]
1，BMES format is automatically supported, use it as your will;
2, char-base and word-based corpus use different formats, it will be detected automatically. Prepare your corpus according to the format, that's all you need to do.

**Comment by marcusau** [comment]
got it ...thanks a lot ... if i change from BI to BMES format.......... my changes are like:

dataset.txt:
市领导到成都...    ["B", "B", "I", "B", "B", "I", ...] --->  ["S", "B", "E", "S", "B", "E", ...]

tags.json:  
["B", "I", ] -->["B", "M", "E", "S"]   # . I skip these two "B-LOC", "I-LOC" as i just use for CWS

vocab.json:
Same as your example


right???  Anything change i am missing???

**Comment by jidasheng** [comment]
yes, it's right. Just try it！

**Comment by marcusau** [comment]
got it .. will try tonight

**Comment by scalleri** [comment]
hey there!

I had to adapt the utils.py script and change the `torch.load(model_path)` to `torch.load(model_path, map_location="cpu")`. So it works on cpu only (after the torch cpu only install). Might be useful for someone! 


```
def build_model(args, processor, load=True, verbose=False):
    model = BiRnnCrf(len(processor.vocab), len(processor.tags),
                     embedding_dim=args.embedding_dim, hidden_dim=args.hidden_dim, num_rnn_layers=args.num_rnn_layers)

    # weights
    model_path = model_filepath(args.model_dir)
    if exists(model_path) and load:
        state_dict = torch.load(model_path, map_location="cpu")
        model.load_state_dict(state_dict)
        if verbose:
            print("load model weights from {}".format(model_path))
    return model

```

have a nice day!
